pkg/minishift/clusterup/clusterup.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterup
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/blang/semver"
+	"github.com/docker/machine/libmachine/host"
+	"github.com/docker/machine/libmachine/provision"
+
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minikube/kubeconfig"
+	"github.com/minishift/minishift/pkg/minishift/addon/command"
+	"github.com/minishift/minishift/pkg/minishift/addon/manager"
+	"github.com/minishift/minishift/pkg/minishift/oc"
+	"github.com/minishift/minishift/pkg/util"
+	"github.com/minishift/minishift/pkg/version"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	ip_key             = "ip"
+	routing_suffix_key = "routing-suffix"
+)
+
+type ClusterUpConfig struct {
+	OpenShiftVersion string
+	MachineName      string
+	Ip               string
+	Port             int
+	RoutingSuffix    string
+	HostPvDir        string
+	User             string
+	Project          string
+	KubeConfigPath   string
+	OcPath           string
+}
+
+// ClusterUp downloads and installs the oc binary in order to run 'cluster up'
+func ClusterUp(config *ClusterUpConfig, clusterUpParams map[string]string, runner util.Runner) error {
+	cmdArgs := []string{"cluster", "up", "--use-existing-config"}
+
+	for key, value := range clusterUpParams {
+		if !oc.SupportFlag(key, config.OcPath, runner) {
+			errors.New(fmt.Sprintf("Flag %s is not supported for oc version %s. Use 'openshift-version' flag to select a different version of OpenShift.", key, config.OpenShiftVersion))
+		}
+		cmdArgs = append(cmdArgs, "--"+key)
+		cmdArgs = append(cmdArgs, value)
+	}
+
+	exitCode := runner.Run(os.Stdout, os.Stderr, config.OcPath, cmdArgs...)
+	if exitCode != 0 {
+		errors.New("Error starting the cluster.")
+	}
+	return nil
+}
+
+// PostClusterUp runs the Minishift specific provisioning after 'cluster up' has run
+func PostClusterUp(clusterUpConfig *ClusterUpConfig, sshCommander provision.SSHCommander, addOnManager *manager.AddOnManager) error {
+	err := kubeconfig.CacheSystemAdminEntries(clusterUpConfig.KubeConfigPath, getConfigClusterName(clusterUpConfig.Ip, clusterUpConfig.Port))
+	if err != nil {
```

Comment:
I am thinking on this line.
```
func checkAndReturnError(err error) {
       if err != nil {
            return err
       }
}

// Call above instead of three lines
```
Choose the name accordingly :smile: